### DISCUSSION TITLE
Load Bitmap Image as Data

### DISCUSSION THREAD

**Discussion by javad0470** [skydoves/landscapist#528] [Q&A]
Hello Dear.
I have question about loading ImageBitmap in my Coil Image loader.
I have image picker that return a file as seleceted image and I convert that to androidx.compoe.ui.graphics.ImageBitmap in Compose Multipatform.
then I pass this image bitmap to imageModel of coil and it doesn't  accept this type of data 
how can I use this type of data in my project?

**Comment by faraz152** [comment]
If you already have an `ImageBitmap` in memory (from a file picker or any other source), you don't need Coil or landscapist at all — just use Compose's built-in `Image` composable directly:

```kotlin
import androidx.compose.foundation.Image
import androidx.compose.ui.graphics.ImageBitmap

@Composable
fun PickedImage(bitmap: ImageBitmap) {
    Image(
        bitmap = bitmap,
        contentDescription = null,
        modifier = Modifier.fillMaxWidth()
    )
}
```

This works identically on Android and iOS in Compose Multiplatform.

---

### If you specifically need CoilImage (e.g. for transformations/caching)

Pass the **file path or URI** rather than the `ImageBitmap`. Coil knows how to load from a `java.io.File`, Android `Uri`, or a Kotlin `okio.Path`:

```kotlin
// Android
CoilImage(
    imageModel = { uri },   // android.net.Uri from the file picker
    modifier = Modifier.fillMaxWidth()
)

// Compose Multiplatform (okio)
CoilImage(
    imageModel = { file },  // java.io.File
    modifier = Modifier.fillMaxWidth()
)
```

Coil does not currently have a built-in `ImageBitmap` `Fetcher`, so there's no way to pass an `ImageBitmap` instance directly to `CoilImage`. The recommended approach is to either keep the file reference and pass that, or use the plain `Image` composable when you already have the decoded bitmap.
